File: go/2020/day9/main_test.go
package main

import (
	"testing"
)

var (
	input = []int{
		35,
		20,
		15,
		25,
		47,
		40,
		62,
		55,
		65,
		95,
		102,
		117,
		150,
		182,
		127,
		219,
		299,
		277,
		309,
		576,
	}
)

func TestPart1(t *testing.T) {
	expected := 127
	actual, _ := part1(input, 5)
	if actual != expected {
		t.Errorf("actual [%v] != expected [%v]", actual, expected)
	}
}

func TestPart2(t *testing.T) {
	expected := 62
	actual := part2(input, 14)
	if actual != expected {
		t.Errorf("actual [%v] != expected [%v]", actual, expected)
	}
}

File: 2022/day14/js/main.test.js
"use strict";

const assert = require("assert");
const { part1, part2 } = require("./main");

const test_data = `498,4 -> 498,6 -> 496,6
503,4 -> 502,4 -> 502,9 -> 494,9`;

async function test_part1(input) {
  const expected = 24;
  const actual = await part1(input);
  assert.equal(
    actual,
    expected,
    `actual: [${actual}] != expected [${expected}]`
  );
}

async function test_part2(input) {
  const expected = 93;
  const actual = await part2(input);
  assert.equal(
    actual,
    expected,
    `actual: [${actual}] != expected [${expected}]`
  );
}

async function test() {
  const input = test_data.split("\n");
  await test_part1(input);
  await test_part2(input);
}

test();
